### DISCUSSION TITLE
harvest 24.11.1 netport.yaml ifgrp query removed

### DISCUSSION THREAD

**Discussion by wagneradrian92** [NetApp/harvest#3338] [General]
hey everyone,

i was just wondering why the query for the ifgrps got removed from the netport.yml config? 
in version 24.5.0 this was still there:

  - query: api/private/cli/network/port/ifgrp
    counters:
      - ^^ifgrp                              => port
      - ^^node                               => node
      - ^activeports                         => activeports
      - ^distr_func                          => ifgrp_func
      - ^down_ports                          => down_ports
      - ^ports                               => ports
      - ^up_ports                            => up_ports
      
is there a new way to collect metrics for ifgrps?

thanks!

regards
adrian

**Comment by Hardikl** [comment]
@wagneradrian92 

Please refer this detail: https://github.com/NetApp/harvest/pull/3117#issuecomment-2312836894

As mentioned in above link, it doesn't seems 1-1 mapping between call (`api/network/ethernet/ports` and `api/private/cli/network/port` - they are fine with same key and can add labels appropriately) and `api/private/cli/network/port/ifgrp` (totally different object call).

Hence, we have decided internally in team to remove this rest call as it's already showing the wrong data. 
If required later, we can add new template for monitoring the ifgrp object separately.

Please open new feature request for monitoring the `ifgrps` object.

**Reply by wagneradrian92** [reply]
@Hardikl 
Thanks for the explanation. i added the query to our config since we are alerting on the "activeports" attribute. waring if partial and alert when both ports of an ifgrp are down.

**Comment by Hardikl** [comment]
@wagneradrian92 ,

Could you please share the response of these below harvest rest cli commands, so we can replay them locally to understand better. Feel free to [email](ng-harvest-files@netapp.com) them if you would prefer not to post them.

**Note:** Replace `POLLER_NAME` with your poller which has the required data of ifgrp.
```
./bin/harvest rest show data -p POLLER_NAME --api network/ethernet/ports > public_ethernet_ports.json
```
```
./bin/harvest rest show data -p POLLER_NAME --api private/cli/network/port --fields node,port,duplex_admin,duplex_oper,flowcontrol_admin,role,speed_admin" > private_ethernet_ports.json
```
```
./bin/harvest rest show data -p POLLER_NAME --api private/cli/network/port/ifgrp  --fields "ifgrp,node,activeports,distr_func,down_ports,ports,up_ports" > private_ifgrps.json
```

Also, If you could provide more information about how the `activeports` label or any other label would be used in your alerts, we would have a better understanding of your use case and provide best solution for it.

**Reply by wagneradrian92** [reply]
@Hardikl,
i sent you the files via email. 
the alert we are currently using looks like this:

alert:NetAppIfgrpHealthDegraded
expr:netapp_net_port_status{activeports="partial",port_type="lag"}

so we simply check if an ifgrp has a degraded state and from there on check on the system where the problem may be. but since an ifgrp is redundant this is no "critical" alert for us so we have enough time to investigate the issue.

**Reply by Hardikl** [reply]
Thanks @wagneradrian92 for the json responses and alert rules about how it's consumed.

We have added the ifgrp api as part of the PR: https://github.com/NetApp/harvest/pull/3342

**Reply by Hardikl** [reply]
Hi @wagneradrian92,
Changes are available in [nightly](https://github.com/NetApp/harvest/releases/tag/nightly). 
Please give it a try when you get a chance and let us know if it solves the issue.
